client/state/comments/actions.js

```diff
@@ -0,0 +1,381 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	COMMENTS_RECEIVE,
+	COMMENTS_COUNT_RECEIVE,
+	COMMENTS_REQUEST,
+	COMMENTS_REQUEST_SUCCESS,
+	COMMENTS_REQUEST_FAILURE,
+	COMMENTS_REMOVE_COMMENT,
+	COMMENTS_ERROR_COMMENT
+} from '../action-types';
+import {
+	createCommentTargetId,
+	createRequestId,
+	createUrlFromTemplate,
+	fieldsMapper
+} from './utils';
+
+const MAX_NUMBER_OF_COMMENTS_PER_FETCH = 50;
+const MAX_NUMBER_OF_COMMENTS_IN_FETCH_RESULT = 100; /* https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/posts/%24post_ID/replies/ */
+
+/***
+ * Internal handler for comments request success
+ * @param {Function} dispatch redux dispatch function
+ * @param {String} requestId request identifier
+ * @param {Number} siteId site identifier
+ * @param {Number} postId post identifier
+ * @param {Object[]} comments comments array
+ * @param {Number} totalCommentsCount comments count that the server have on that post
+ */
+function commentsRequestSuccess( dispatch, requestId, siteId, postId, comments, totalCommentsCount ) {
+	dispatch( {
+		type: COMMENTS_REQUEST_SUCCESS,
+		requestId: requestId
+	} );
+
+	dispatch( {
+		type: COMMENTS_RECEIVE,
+		siteId: siteId,
+		postId: postId,
+		comments: comments
+	} );
+
+	// if the api have returned comments count, dispatch it
+	if ( totalCommentsCount > -1 ) {
+		dispatch( {
+			type: COMMENTS_COUNT_RECEIVE,
+			siteId,
+			postId,
+			totalCommentsCount
+		} );
+	} else {
+		requestPostCommentsCount( siteId, postId )( dispatch );
+	}
+}
+
+/***
+ * Internal handler for comments request failure
+ * @param {Function} dispatch redux dispatch function
+ * @param {String} requestId request identifier
+ * @param {Object} err error object
+ */
+function commentsRequestFailure( dispatch, requestId, err ) {
+	dispatch( {
+		type: COMMENTS_REQUEST_FAILURE,
+		requestId: requestId,
+		error: err
+	} );
+
+	throw err;
```

Comment:
Since it's just a helper function for `requestPostComments` If I don't throw here, I silently swallow the error, it won't propagate further to whoever gets the resulting promise from `requestPostComments`.